WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m

```diff
@@ -0,0 +1,276 @@
+#import "MediaService.h"
+#import "Blog.h"
+#import "ContextManager.h"
+
+@interface  MediaLibraryPickerDataSource()
+
+@property (nonatomic, strong) MediaLibraryGroup *mediaGroup;
+@property (nonatomic, strong) Blog *blog;
+@property (nonatomic, assign) WPMediaType filter;
+@property (nonatomic, strong) NSArray *media;
+
+@end
+
+@implementation MediaLibraryPickerDataSource
+
+- (instancetype)initWithBlog:(Blog *)blog 
+{
+    self = [super init];
+    if (self) {
+        _mediaGroup = [[MediaLibraryGroup alloc] initWithBlog:blog];
+        _blog = blog;
+    }
+    return self;
+}
+
+- (instancetype)init
+{
+    return [self initWithBlog:nil];
+}
+
+
+-(NSInteger)numberOfAssets
+{
+    return [self.media count];
+}
+
+-(NSInteger)numberOfGroups
+{
+    return 1;
+}
+
+-(void)setSelectedGroup:(id<WPMediaGroup>)group
+{
+    //There is only one group in the media library for now so don't do anything
+}
+
+-(id<WPMediaGroup>)selectedGroup
+{
+   //There is only one group in the media library for now so don't do anything
+   return self.mediaGroup;
+}
+
+- (id<WPMediaGroup>)groupAtIndex:(NSInteger)index
+{
+    return self.mediaGroup;
+}
+
+-(void)loadDataWithSuccess:(WPMediaChangesBlock)successBlock failure:(WPMediaFailureBlock)failureBlock
+{
+    NSString *entityName = NSStringFromClass([Media class]);
+    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:entityName];
+    request.predicate = [[self class] predicateForFilter:self.filter];
+    NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:YES];
+    request.sortDescriptors = @[sortDescriptor];
+    NSError *error;
+    self.media = [[[ContextManager sharedInstance] mainContext] executeFetchRequest:request error:&error];
+    if (self.media == nil && error){
+        DDLogVerbose(@"Error fecthing media: %@", [error localizedDescription]);
+        if (failureBlock) {
+            failureBlock(error);
+        }
+        return;
+    }
+    if (successBlock) {
+        successBlock();
+    }
+}
+
+-(id<NSObject>)registerChangeObserverBlock:(WPMediaChangesBlock)callback
+{
+    return nil;
+}
+
+-(void)unregisterChangeObserver:(id<NSObject>)blockKey
+{
+
+}
+
+-(void)addImage:(UIImage *)image metadata:(NSDictionary *)metadata completionBlock:(WPMediaAddedBlock)completionBlock
+{
+
+}
+
+-(void)addVideoFromURL:(NSURL *)url completionBlock:(WPMediaAddedBlock)completionBlock
+{
+
+}
+
+-(void)setMediaTypeFilter:(WPMediaType)filter
+{
+    self.filter = filter;
+    self.mediaGroup.filter = filter;
+}
+
+-(WPMediaType)mediaTypeFilter
+{
+    return self.filter;
+}
+
+-(id<WPMediaAsset>)mediaAtIndex:(NSInteger)index
+{
+    return self.media[index];
+}
+
++ (NSPredicate *)predicateForFilter:(WPMediaType)filter
+{
+    NSPredicate *predicate;
+    switch (filter){
+        case WPMediaTypeImage:{
+            predicate = [NSPredicate predicateWithFormat:@"mediaTypeString = %@  && mediaID != 0", @"image"];
+        }break;
```

Comment:
Not spelled out in the style guide...but should break be on a new line? I'll leave that up to you ;-)